src/data/matrix/rank.lean

```diff
@@ -154,4 +160,105 @@ lemma rank_eq_finrank_span_cols (A : matrix m n R) :
   A.rank = finrank R (submodule.span R (set.range Aᵀ)) :=
 by rw [rank, matrix.range_mul_vec_lin]
 
+end comm_ring
+
+/-! ### Lemmas about transpose and conjugate transpose
+
+This section contains lemmas about the rank of `matrix.transpose` and `matrix.conj_transpose`.
+
+Unfortunately the proofs are essentially duplicated between the two; `ℚ` is a linearly-ordered ring
+but can't be a star-ordered ring, while `ℂ` is star-ordered (with `open_locale complex_order`) but
+not linearly ordered. For now we don't prove the transpose case for `ℂ`.
+-/
+
+section star_ordered_field
+variables [fintype m] [field R] [partial_order R] [star_ordered_ring R]
+
+lemma ker_mul_vec_lin_conj_transpose_mul_self (A : matrix m n R) :
+  linear_map.ker (Aᴴ ⬝ A).mul_vec_lin = linear_map.ker (mul_vec_lin A):=
```

Comment:
Not directly related to this PR, but I find that `mul_vec_lin` instead of `to_lin` is really obscure here. To avoid the decidable equality issue, I'd rename the current `to_lin` as `equiv_lin` and use `to_lin` for, well, turning a matrix into a linear map (i.e., the current `mul_vec_lin`)...

Reply:
If we want to do that it should be ASAP, as the file defining `to_lin` is almost ready to port.

The header comment is already full of refactor ideas corresponding to non-commutative rings, suggesting that we maybe have picked the wrong operator for `to_lin`...

Reply:
I think it's better to keep this for after the port, no need to delay things unnecessarily.